indexing-service/src/main/java/io/druid/indexing/overlord/HeapMemoryTaskStorage.java

```diff
@@ -198,6 +197,30 @@ public void addLock(final String taskid, final TaskLock taskLock)
   }
 
   @Override
+  public void upgradeLock(String taskid, TaskLock taskLock)
```

Comment:
Oh..I see what you are saying. Sorry I could not get what you were saying earlier. 
Yes, it already gets the `taskLock` that user want to have, it exactly does what you are suggesting